### ISSUE TITLE
COMPILE-FILE-ERROR on installation (as dependency of linedit)

### ISSUE THREAD

**Issue by marsjaninzmarsa** [osicat/osicat#14]
SBCL 1.2.4.debian (on Jessie), two machines, one with upgraded packages and one fresh install.

```
ASDF could not load linedit because Error while trying to load definition for system osicat from pathname /home/marsjan/quicklisp/dists/quicklisp/software/osicat-20160421-git/osicat.asd: COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "cffi-toolchain" "toolchain" "asdf-compat">.

debugger invoked on a LOAD-SYSTEM-DEFINITION-ERROR in thread #<THREAD "main thread" RUNNING {10039CE5A3}>: Error while trying to load definition for system osicat from pathname /home/marsjan/quicklisp/dists/quicklisp/software/osicat-20160421-git/osicat.asd: COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "cffi-toolchain" "toolchain" "asdf-compat">

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [RETRY                        ] Retry compiling #<CL-SOURCE-FILE "cffi-toolchain" "toolchain" "asdf-compat">.
  1: [ACCEPT                       ] Continue, treating compiling #<CL-SOURCE-FILE "cffi-toolchain" "toolchain" "asdf-compat"> as having been successful.
  2: [RETRY                        ] Retry EVAL of current toplevel form.
  3: [CONTINUE                     ] Ignore error and continue loading file "/home/marsjan/quicklisp/dists/quicklisp/software/osicat-20160421-git/osicat.asd".
  4: [ABORT                        ] Abort loading file "/home/marsjan/quicklisp/dists/quicklisp/software/osicat-20160421-git/osicat.asd".
  5:                                 Retry ASDF operation.
  6: [CLEAR-CONFIGURATION-AND-RETRY] Retry ASDF operation after resetting the configuration.
  7:                                 Retry EVAL of current toplevel form.
  8:                                 Ignore error and continue userinit file "/home/marsjan/.sbclrc".
  9:                                 Abort userinit file "/home/marsjan/.sbclrc".
 10:                                 Skip to toplevel READ/EVAL/PRINT loop.
 11: [EXIT                         ] Exit SBCL (calling #'EXIT, killing the process).

((FLET #:HANDLER1584 :IN ASDF/FIND-SYSTEM:LOAD-ASD) #<COMPILE-FILE-ERROR {100C547FB3}>)
```

**Comment by luismbo** [comment]
Looks like an ASDF/CFFI issue. What version of ASDF and CFFI are you using?

**Comment by micrub** [comment]
As hinted before, same error resolved on Ubuntu 14.04 by actually installing ASDF using quicklisp and `sudo apt-get install cl-cffi -y`.

Thou still generates following error:

```
(ql:quickload "linedit")
To load "linedit":
  Load 1 ASDF system:
    linedit
; Loading "linedit"
[package alexandria.0.dev]........................
[package babel-encodings].........................
[package babel]...................................
[package cffi-sys]................................
[package cffi]....................................
[package cffi-features]...........................
[package cffi-grovel].............................
[package uffi]....................................
[package osicat-sys]..............................
[package osicat-posix]; cc -m64 -I/usr/share/common-lisp/source/cl-cffi/ -fPIC -o /home/.cache/common-lisp/sbcl-1.1.14.debian-linux-x64/home/quicklisp/dists/quicklisp/software/osicat-20160825-git/posix/basic-unixint /home/.cache/common-lisp/sbcl-1.1.14.debian-linux-x64/home/quicklisp/dists/quicklisp/software/osicat-20160825-git/posix/basic-unixint.c
; /home/.cache/common-lisp/sbcl-1.1.14.debian-linux-x64/home/quicklisp/dists/quicklisp/software/osicat-20160825-git/posix/basic-unixint /home/.cache/common-lisp/sbcl-1.1.14.debian-linux-x64/home/quicklisp/dists/quicklisp/software/osicat-20160825-git/posix/basic-unixint.grovel-tmp.lisp
..
debugger invoked on a TYPE-ERROR in thread #<THREAD "main thread" RUNNING {1002A8B3F3}>: The value (:UNION SIGVAL) is not of type SYMBOL.
```

**Comment by luismbo** [comment]
Try to `apt-get remove` cl-cffi and let quicklisp download it instead.
